cmd/run-once-duration-override/runoncedurationoverride.go

```diff
@@ -46,6 +53,21 @@ func (m *runOnceDurationOverrideHook) Initialize(kubeClientConfig *restclient.Co
 
 	klog.V(1).Infof("name=%s admission webhook loaded successfully", runoncedurationoverride.Name)
 
+	var LoopbackClientConfig *restclient.Config
+	var SecureServing *apiserver.SecureServingInfo
+
+	secureServing := apiserveroptions.NewSecureServingOptions().WithLoopback()
+	secureServing.BindPort = DefaultRODOMetricsPort
+
+	if err := secureServing.ApplyTo(&SecureServing, &LoopbackClientConfig); err != nil {
+		klog.V(1).Infof("name=%s failed to initialize webhook mectrics - %s", runoncedurationoverride.Name, err.Error())
+		return err
+	}
+	pathRecorderMux := mux.NewPathRecorderMux("rodo")
+	pathRecorderMux.Handle("/metrics", legacyregistry.HandlerWithReset())
```

Comment:
```go
_, _, err = SecureServing.Serve(pathRecorderMux, 0, stopCh)
if err != nil {
        klog.Fatalf("failed to start secure server: %v", err)
        return err
}
```
is missing. Once added, the metrics are published and accessible through `/metrics` endpoint

Reply:
Worth moving the code from the `Initialize` method.

Reply:
We use the stopCh and we return errors here. Wouldn't it be better to leave it in Initialize?

Reply:
Worth considering to either refactor https://github.com/openshift/run-once-duration-override/blob/main/vendor/github.com/openshift/generic-admission-server/pkg/cmd/cmd.go or inline it instead of directly invoking `RunAdmissionServer`. E.g.
```
func RunAdmissionServer(admissionHooks ...AdmissionHook) {
	if len(os.Getenv("GOMAXPROCS")) == 0 {
		runtime.GOMAXPROCS(runtime.NumCPU())
	}

	stopCh := genericapiserver.SetupSignalHandler()

	<< inject the metrics initialization here

	// done to avoid cannot use admissionHooks (type []AdmissionHook) as type []apiserver.AdmissionHook in argument to "github.com/openshift/kubernetes-namespace-reservation/pkg/genericadmissionserver/cmd/server".NewCommandStartAdmissionServer
	var castSlice []apiserver.AdmissionHook
	for i := range admissionHooks {
		castSlice = append(castSlice, admissionHooks[i])
	}

	code := cli.Run(server.NewCommandStartAdmissionServer(os.Stdout, os.Stderr, stopCh, castSlice...))
	os.Exit(code)
}
```